### DISCUSSION TITLE
NOTESPALSHES FIANILLY FOR HOW TO ADD KADE ENGINE 1.6.2 (FIXED?)

### DISCUSSION THREAD

**Discussion by glowsoony** [KadeArchive/Kade-Engine#2655] [General]
I FOUND HOW TO ADD IT.
![image](https://user-images.githubusercontent.com/84847356/139561347-d1903074-e710-4aae-95f9-6aba7d4d0d72.png)

find the lines were in playstate.hx

		var daRating = daNote.rating;

		switch (daRating)
		{
			case 'shit':
				score = -300;
				combo = 0;
				misses++;
				health -= 0.1;
				ss = false;
				shits++;
				if (FlxG.save.data.accuracyMod == 0)
					totalNotesHit -= 1;
			case 'bad':
				daRating = 'bad';
				score = 0;
				health -= 0.06;
				ss = false;
				bads++;
				if (FlxG.save.data.accuracyMod == 0)
					totalNotesHit += 0.50;
			case 'good':
				daRating = 'good';
				score = 200;
				ss = false;
				goods++;
				if (FlxG.save.data.accuracyMod == 0)
					totalNotesHit += 0.75;
			case 'sick':
				if (health < 2)
					health += 0.04;
				if (FlxG.save.data.accuracyMod == 0)
					totalNotesHit += 1;
				sicks++;
		}
and add a this BUT REMOVE THE IF (FlxG.save.data.Notesplash)

![image](https://user-images.githubusercontent.com/84847356/165859267-e340940d-6a6d-4544-96f0-549b4879968e.png)

also set the sploosh.alpha = 0.6; NOT FlxG.save.data.notesplashalpha

var sploosh:FlxSprite = new FlxSprite(daNote.x, playerStrums.members[daNote.noteData].y);
			{
				var tex:flixel.graphics.frames.FlxAtlasFrames = Paths.getSparrowAtlas('noteSplashes', 'shared');
				sploosh.frames = tex;
				sploosh.animation.addByPrefix('splash 0 0', 'note impact 1 purple', 24, false);
				sploosh.animation.addByPrefix('splash 0 1', 'note impact 1  blue', 24, false);
				sploosh.animation.addByPrefix('splash 0 2', 'note impact 1 green', 24, false);
				sploosh.animation.addByPrefix('splash 0 3', 'note impact 1 red', 24, false);
				sploosh.animation.addByPrefix('splash 1 0', 'note impact 2 purple', 24, false);
				sploosh.animation.addByPrefix('splash 1 1', 'note impact 2 blue', 24, false);
				sploosh.animation.addByPrefix('splash 1 2', 'note impact 2 green', 24, false);
				sploosh.animation.addByPrefix('splash 1 3', 'note impact 2 red', 24, false);
                
				if (daRating == 'sick')
				{
					add(sploosh);
					sploosh.cameras = [camHUD];
					sploosh.animation.play('splash ' + FlxG.random.int(0, 1) + " " + daNote.noteData);
					sploosh.alpha = 0.6;
					sploosh.offset.x += 90;
					sploosh.offset.y += 80;
					sploosh.animation.finishCallback = function(name) sploosh.kill();
				}
			}
and 
![noteSplashes](https://user-images.githubusercontent.com/84847356/139591546-376a1644-969d-42de-ab37-778d3b8bccb3.png)
download that and paste it into shared/images

![image](https://user-images.githubusercontent.com/84847356/139591573-77a9055b-63df-4cfa-b241-b7936b727387.png)

now run your game.

now make a xml and name it noteSplashes and it should be named noteSplashes.xml

and paste this code into it 
https://pastebin.com/aiYVqpaT

**Comment by glowsoony** [comment]
IF ANY OF YALL SAW THIS BEFORE I CHANGED THE CODE THEN I AM VERY SORRY
BUT I FOUND THAT AN ERROR WAS MESSING TO NOT PLAY THE ANIMTIONS FOR PIXEL LEVELS SO I FIXED THE CODE NOW
SORRY YALL BUT IF YOU CAME AND FIX IT THANK YOU!

**Reply by glowsoony** [reply]
this should fix it or not?

**Comment by AwesomeWithRex** [comment]
In pixel level when you press the down note it stays there

**Reply by glowsoony** [reply]
> In pixel level when you press the down note it stays there

thats odd.

**Comment by RealDealJinx** [comment]
imma test this to see if it works on 1.8

**Reply by RealDealJinx** [reply]
Works on 1.8 but the down note stays there

**Reply by RealDealJinx** [reply]
There is a fix but i found it within the friday night fever code

check these 2
https://github.com/zacksgamerz/Friday-Night-Fever/blob/main/source/PlayState.hx
https://github.com/zacksgamerz/Friday-Night-Fever/blob/main/source/NoteSplash.hx

**Comment by BinaricMC** [comment]
does it work on 1.8?

**Reply by glowsoony** [reply]
yes all versions

**Reply by BinaricMC** [reply]
> yes all versions

shows this 
![image](https://user-images.githubusercontent.com/76714822/166166678-e8b05f38-ebe2-4a8d-a47f-0e752295e590.png)

**Reply by glowsoony** [reply]
> > yes all versions
> 
> shows this ![image](https://user-images.githubusercontent.com/76714822/166166678-e8b05f38-ebe2-4a8d-a47f-0e752295e590.png)

thats weird

**Reply by glowsoony** [reply]
maybe it dose not work in 1.8 as i thought it would :(

**Reply by BinaricMC** [reply]
> maybe it dose not work in 1.8 as i thought it would :(

i got it work

**Comment by RobGames678** [comment]
Look, I DON’T WANNA KADE ENGINE MESSAGES ANYMORE !!! Sent from Mail for Windows 10 From: MinecraftBoy2038Sent: Friday, May 6, 2022 6:04 PMTo: KadeDev/Kade-EngineCc: SubscribedSubject: Re: [KadeDev/Kade-Engine] NOTESPALSHES FIANILLY FOR HOW TO ADD KADE ENGINE 1.6.2 (FIXED?) (Discussion #2655) maybe it dose not work in 1.8 as i thought it would :(i got it work—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>
